packages/postgrest-core/src/update-fetcher.ts

```diff
@@ -35,17 +37,25 @@ export const buildUpdateFetcher =
   >(
     qb: PostgrestQueryBuilder<S, T, Re>,
     primaryKeys: (keyof T['Row'])[],
-    opts: BuildNormalizedQueryOps<Q> & UpdateFetcherOptions<S, T>,
+    {
+      stripPrimaryKeys = true,
+      ...opts
+    }: BuildNormalizedQueryOps<Q> & UpdateFetcherOptions<S, T>,
   ): UpdateFetcher<T, R> =>
   async (
     input: Partial<T['Row']>,
   ): Promise<MutationFetcherResponse<R> | null> => {
-    let filterBuilder = qb.update(input as any, opts); // todo fix type;
+    const payload = { ...input };
+    let filterBuilder = qb.update(payload as any, opts); // todo fix type;
     for (const key of primaryKeys) {
       const value = input[key];
       if (!value)
         throw new Error(`Missing value for primary key ${String(key)}`);
       filterBuilder = filterBuilder.eq(key as string, value);
+
+      if (stripPrimaryKeys) {
+        payload[key] = undefined;
+      }
```

Comment:
should this not happen before we pass the payload to `qb.update`?

Reply:
It doesn't since we don't call the query before, and qb.update just sets a stable reference to the payload object. Thought it would keep the code simpler for future changes if we do related manipulations in one pass.

It should have tests though, I agree with that.

Reply:
I would prefer to move it up - makes it simpler to read and there is no other reason to put it down.